h/static/scripts/group-forms/components/CreateGroupForm.tsx

```diff
@@ -122,13 +122,26 @@ export default function CreateGroupForm() {
   const [name, setName] = useState('');
   const [description, setDescription] = useState('');
   const [errorMessage, setErrorMessage] = useState('');
+  const [saving, setSaving] = useState(false);
   const config = useMemo(() => readConfig(), []);
 
+  useEffect(() => {
+    window.addEventListener('pageshow', e => {
+      if (e.persisted) {
+        // Disable the "saving" state if the user uses the browser's Back button to navigate
+        // back to the group form.
+        setSaving(false);
+      }
+    });
+  }, [setSaving]);
```

Comment:
This should be:

```js
  useEffect(() => {
    const listener = (e: PageTransitionEvent) => {
      if (e.persisted) {
        // Disable the "saving" state if the user uses the browser's Back button to navigate
        // back to the group form.
        setSaving(false);
      }
    };
    window.addEventListener('pageshow', listener);
    return () => window.removeEventListener('pageshow', listener);
  }, []);
```

- State setters don't need to be listed in the dependencies array, as they are immutable
- Effects that subscribe to events etc. should return a cleanup function that does the effects of the listener. That won't actually be used in the application, but will ensure that the listeners can be removed in tests.

In tests you'll need to unmount the component after each test to avoid this listener remaining around. See https://github.com/hypothesis/via/blob/d11083d6b8d1bf85e10407278018ba44f8147f58/via/static/scripts/video_player/components/test/HTMLVideoPlayer-test.js#L5-L20 for a pattern that we use for this.

Reply:
:+1: Done